src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -31,44 +34,66 @@ import { ConfigService } from "../../config/config.service";
   templateUrl: "./navigation.component.html",
   styleUrls: ["./navigation.component.scss"],
 })
+@UntilDestroy()
 export class NavigationComponent {
+  activeElement: string;
   /** name of config array in the config json file */
   private readonly CONFIG_ID = "navigationMenu";
   /** all menu items to be displayed */
   public menuItems: MenuItem[] = [];
 
   constructor(
     private adminGuard: AdminGuard,
-    private configService: ConfigService
+    private configService: ConfigService,
+    private router: Router
   ) {
     this.configService.configUpdated.subscribe(() =>
       this.initMenuItemsFromConfig()
     );
+    this.router.events
+      .pipe(
+        startWith(new NavigationEnd(0, this.router.url, "")),
+        filter((event) => event instanceof NavigationEnd),
+        map((event: NavigationEnd) =>
+          // conservative filter matching all items that could fit to the given url
+          this.menuItems.filter((item) => event.url.startsWith(item.link))
+        )
+      )
+      .subscribe((items) => {
+        if (items.length === 0) {
+          this.activeElement = "";
+        } else if (items.length === 1) {
+          this.activeElement = items[0].name;
+        } else {
+          // If there are multiple matches (multiple elements start with the give link),
+          // use the element where the length is bigger. For example:
+          // '/attendance' matches both manage attendance ('/attendance') as well as
+          // '/attendance/add/day' (add day attendance).
+          // If 'attendance/add/day' was the active link, then only this link
+          // would have been filtered
+          this.activeElement = items.reduce((i1, i2) =>
+            i1.link.length > i2.link.length ? i1 : i2
+          ).name;
+        }
+      });
   }
 
   /**
    * Load menu items from config file
    */
   private initMenuItemsFromConfig() {
-    this.menuItems = [];
-    const config: NavigationMenuConfig = this.configService.getConfig<NavigationMenuConfig>(
+    const config = this.configService.getConfig<NavigationMenuConfig>(
       this.CONFIG_ID
     );
-    for (const configItem of config.items) {
-      if (this.checkMenuItemPermissions(configItem.link)) {
-        this.menuItems.push(
-          new MenuItem(configItem.name, configItem.icon, configItem.link)
-        );
-      }
-    }
+    this.menuItems = config.items.filter((item) => this.hasPermissionFor(item));
   }
 
   /**
    * Check whether the user has the required rights
    */
-  private checkMenuItemPermissions(link: string): boolean {
+  private hasPermissionFor(item: MenuItem): boolean {
```

Comment:
using this function (`this.hasPermissionFor(item)`) sounds a little bit like if the function would check if the current component has the required permissions. 
```suggestion
  private currentUserHasPermissionsFor(item: MenuItem): boolean {
```
or
```suggestion
  private currentUserIsPermitted(item: MenuItem): boolean {
```